Source/carthage/Formatting.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  Formatting.swift
+//  Carthage
+//
+//  Created by J.D. Healy on 1/29/15.
+//  Copyright (c) 2015 Carthage. All rights reserved.
+//
+
+import Commandant
+import Foundation
+import LlamaKit
+import PrettyColors
+import ReactiveCocoa
+
+func wrap(colorful: Bool)(wrap: Color.Wrap)(string: String) -> String {
+	return colorful ? wrap.wrap(string) : string
+}
+
+internal struct Terminal {
+	static var term: String? {
+		return getEnvironmentVariable("TERM").value()
+	}
+	static var isDumb: Bool {
+		return (Terminal.term?.lowercaseString as NSString?)?.isEqualToString("dumb") ?? false
+	}
+	static var isTTY: Bool {
+		return isatty(STDOUT_FILENO) == 1
+	}
+}
+
+public enum ColorArgument: String, ArgumentType, Printable {
+	case Auto = "auto"
+	case Never = "never"
+	case Always = "always"
+	
+	public var isColorful: Bool {
+		switch self {
+		case .Always:
+			return true
+		case .Never:
+			return false
+		case .Auto:
+			return Terminal.isTTY && !Terminal.isDumb
+		}
+	}
+	
+	public var description: String {
+		return "‘auto’ || ‘always’ || ‘never’"
```

Comment:
Yeah, changed this because in earlier commits `carthage help xxxxx` would have output like
`[--color (Enum value)]`


Looking at `carthage help checkout`'s output…
```
[--configuration Release]
the Xcode configuration to build (ignored if --no-build option is present)
```
… Commandant seems to print `[``--key` `defaultValue.description``]`-type output.

I guess I should move the `‘auto’ || ‘always’ || ‘never’` part back to `usage` [like it was here](https://github.com/jdhealy/Carthage/commit/eec90adec02ff99b2fb6d3530af4c60bd278b3f2) and change the description to the `rawValue`. Good catch :+1: